app/src/ui/multi-commit-operation/choose-branch/base-choose-branch-dialog.tsx

```diff
@@ -65,87 +89,39 @@ interface IBaseChooseBranchDialogProps {
   readonly onDismissed: () => void
 }
 
-export interface IBaseChooseBranchDialogState {
-  /** The currently selected branch. */
+export interface IChooseBranchDialogProps extends IBaseChooseBranchDialogProps {
   readonly selectedBranch: Branch | null
+  readonly dialogTitle: string | JSX.Element | undefined
+  readonly submitButtonTooltip?: string
+  readonly canStartOperation: boolean
+  readonly start: () => void
+  readonly onSelectionChanged: (selectedBranch: Branch | null) => void
+}
 
+export interface IChooseBranchDialogState {
   /** The filter text to use in the branch selector */
   readonly filterText: string
-
-  /**
-   * A preview of the operation using the selected base branch to test whether the
-   * current branch will be cleanly applied.
-   */
-  readonly statusPreview: JSX.Element | null
 }
 
-export abstract class BaseChooseBranchDialog extends React.Component<
-  IBaseChooseBranchDialogProps,
-  IBaseChooseBranchDialogState
+export class ChooseBranchDialog extends React.Component<
+  IChooseBranchDialogProps,
+  IChooseBranchDialogState
 > {
-  protected abstract start: () => void
-
-  protected abstract canStart: () => boolean
-
-  protected abstract updateStatus: (branch: Branch) => Promise<void>
-
-  protected abstract getDialogTitle: (
-    branchName: string
-  ) => string | JSX.Element | undefined
-
-  protected abstract renderActionStatusIcon: () => JSX.Element | null
-
-  public constructor(props: IBaseChooseBranchDialogProps) {
+  public constructor(props: IChooseBranchDialogProps) {
     super(props)
 
-    const selectedBranch = this.resolveSelectedBranch()
-
     this.state = {
-      selectedBranch,
       filterText: '',
-      statusPreview: null,
     }
   }
 
-  public componentDidMount() {
-    const { selectedBranch } = this.state
-    if (selectedBranch !== null) {
-      this.updateStatus(selectedBranch)
+  public componentDidMount(): void {
+    const initialSelectedBranch = this.resolveSelectedBranch()
+    if (initialSelectedBranch !== null) {
```

Comment:
Done. On component mounting.. it should be null as there is no initial selection and the purpose of the `resolveSelectedBranch` is to determine it's initial selection ... but, I can see as a precaution to do so incase the consumer sets otherwise.